src/services/event/event-router.test.ts

```diff
@@ -0,0 +1,244 @@
+    });
+
+    it("successfully filters and returns a public event for staff", async () => {
+        const eventId = PUBLIC_METADATA.eventId;
+        const response = await getAsStaff(`/event/${eventId}`).expect(StatusCode.SuccessOK);
+        expect(JSON.parse(response.text)).toMatchObject(INTERNAL_PUBLIC_EVENT);
+    });
+
+    it("successfully filters and returns a public event for attendees", async () => {
+        const eventId = PUBLIC_METADATA.eventId;
+        const response = await getAsAttendee(`/event/${eventId}`).expect(StatusCode.SuccessOK);
+        expect(JSON.parse(response.text)).toMatchObject(EXTERNAL_PUBLIC_EVENT);
+    });
+});
+
+describe("GET /event/metadata/:EVENTID", () => {
+    it("cannot be accessed by a non-staff user", async () => {
+        const eventId = PUBLIC_METADATA.eventId;
+        const response = await getAsAttendee(`/event/metadata/${eventId}`).expect(StatusCode.ClientErrorForbidden);
+        expect(response).toHaveProperty("error");
+    });
+
+    it("returns metadata for a particular eventId", async () => {
+        const eventId = PUBLIC_METADATA.eventId;
+        const response = await getAsStaff(`/event/metadata/${eventId}`).expect(StatusCode.SuccessOK);
+        expect(JSON.parse(response.text)).toMatchObject(PUBLIC_METADATA);
+    });
+
+    it("throws an error if a bad eventId is passed in", async () => {
+        const eventId = "badEventId";
+        const response = await getAsStaff(`/event/metadata/${eventId}`).expect(StatusCode.ClientErrorBadRequest);
+        expect(JSON.parse(response.text)).toHaveProperty("error");
+    });
+});
+
+describe("DELETE /event/:EVENTID", () => {
+    it("cannot be accessed by an attendee", async () => {
+        const eventId = "deleteEventId";
+        const response = await delAsAttendee(`/event/${eventId}`).expect(StatusCode.ClientErrorForbidden);
+        expect(response).toHaveProperty("error");
+    });
+
+    it("cannot be accessed by staff", async () => {
+        const eventId = "deleteEventId";
+        const response = await delAsStaff(`/event/${eventId}`).expect(StatusCode.ClientErrorForbidden);
+        expect(response).toHaveProperty("error");
+    });
+
+    it("deletes the events correctly from both tables", async () => {
+        const eventId = "deleteEventId";
+        await delAsAdmin(`/event/${eventId}`).expect(StatusCode.SuccessNoContent);
+
+        const metadata = await Models.EventMetadata.findOne({ eventId: eventId });
+        const event = await Models.PublicEvent.findOne({ eventId: eventId });
+
+        expect(metadata).toBeNull();
+        expect(event).toBeNull();
+    });
+});
+
+describe("PUT /event/metadata/", () => {
+    it("cannot be accessed by attendee", async () => {
+        const response = await putAsAttendee(`/event/metadata/`).expect(StatusCode.ClientErrorForbidden);
+        expect(response).toHaveProperty("error");
+    });
+
+    it("cannot be accessed by staff", async () => {
```

Comment:
don't need to test for every level - just test the level below what is expect (staff for admin only, attendee for staff only, etc)